types/options.go

```diff
@@ -114,11 +120,102 @@ func loadDefaultStoreOptions() {
 
 // loadStoreOptions returns the default storage ops for containers
 func loadStoreOptions() (StoreOptions, error) {
-	storageConf, err := DefaultConfigFile()
+	baseConf, err := DefaultConfigFile()
+	if err != nil {
+		return defaultStoreOptions, err
+	}
+
+	// Load the base config file
+	baseOptions, err := loadStoreOptionsFromConfFile(baseConf)
+	if err != nil {
+		return defaultStoreOptions, err
+	}
+
+	if _, err := os.Stat(defaultDropInConfigDir); err != nil && os.IsNotExist(err) {
+		return defaultStoreOptions, err
+	}
+
+	baseOptions, err = mergeConfigFromDirectory(baseOptions, defaultDropInConfigDir)
 	if err != nil {
 		return defaultStoreOptions, err
 	}
-	return loadStoreOptionsFromConfFile(storageConf)
+
+	return baseOptions, nil
+}
+
+func mergeConfigFromDirectory(baseOptions StoreOptions, configDir string) (StoreOptions, error) {
+	err := filepath.Walk(configDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+
+		// Load drop-in options from the current file
+		dropInOptions, err := loadStoreOptionsFromConfFile(path)
+		if err != nil {
+			return err
+		}
+
+		// Merge the drop-in options into the base options
+		baseOptions = mergeStoreOptions(baseOptions, dropInOptions)
+		return nil
+	})
+
+	if err != nil {
+		return baseOptions, err
+	}
+	return baseOptions, nil
+}
+
+func mergeStoreOptions(base, dropIn StoreOptions) StoreOptions {
+	if dropIn.RunRoot != "" {
```

Comment:
instead of individually adding fields (which won't scale well, can we just serialize into the struct: https://github.com/cri-o/cri-o/blob/256fda5ac98de1d67e26133d0b25df2a74e2ebd2/pkg/config/config.go#L705

Reply:
Thanks. Turns out they already have a function that does that, https://github.com/containers/storage/blob/1fd0dc1d20e523f54369e874d510bf62a23f1574/types/options.go#L415

Reply:
Not really, they don't serialize and automatically update.

Reply:
@haircommander I updated the code to reuse existing [ReloadConfigurationFile](https://github.com/containers/storage/blob/1fd0dc1d20e523f54369e874d510bf62a23f1574/types/options.go#L415C6-L415C29) (instead of writing a new merge config logic) which seem to be doing [more](https://github.com/containers/storage/blob/1fd0dc1d20e523f54369e874d510bf62a23f1574/types/options.go#L446) than just updating the options. `ReloadConfigurationFile` is in use already, so I think we can safely rely on it. Although I admit it is not as scalable as the cri-o approach is, but using `ReloadConfigurationFile` allows us to reuse their existing way of reading and parsing a config file. 

WDYT?